### PR TITLE
Converts screwdrivers to GAS

### PR DESCRIPTION
**Pull request by Arkatos1**
## About The Pull Request

Screwdrivers were converted to the Greyscale Autogenerated Sprites (GAS) system. No sprite changes, as colors and overlays are the same. However this does mean that when you view the item in the shift-right click menu, you will see the actually recolored sprite instead of the default base one.

ScrewdriverGAS

Special attention in review should be given to inhand and worn belt overlays, as that is a new area for GAS sprites.

## Why It's Good For The Game

Cleaner code.

## Changelog
:cl: Arkatos
code: Screwdrivers now use a new greyscaling method for their sprites.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
https://www.youtube.com/watch?v=LVvNme-3ad8

**Comment by maxymax13** [comment]
i cannot read "GAS" as anything other than [gas](https://www.youtube.com/watch?v=LVvNme-3ad8)

**Comment by ninjanomnom** [comment]
I have some code in the works for making proper full icon files using GAS with multiple icon states that would be perfect for the inhand sprites here. I'll split it off from the other code so you can use it here.

**Comment by ninjanomnom** [comment]
You're going to want to make use of #58487 when it's merged instead of the current code you have to choose icons without an icon state.

**Comment by Arkatos1** [comment]
Not sure how to make use of it as I currently understand it - 
1) there are icon variants for right and left inhands, but only one inhand_icon to use in my greyscale config.
2) inhand icons are in different files, so I need to specify them in a different greyscale config anyway if I do not move them into one file

**Comment by celotajstg** [comment]
Figured my PR could be a good reference for inhand adjustments - #58514

**Comment by Arkatos1** [comment]
Appreciate it. Commited an updated version including newest GAGS changes.

**Comment by celotajstg** [comment]
So I noticed in my PR that having a null icon state can lead to an item not being throwable, please check if this affects your PR as well. I resolved this by making use of named states.

Also managed to resolve an issue that prevented me from using GAGS icons in vendors.

**Comment by Arkatos1** [comment]
No such problems on my end luckily.

File: code/game/objects/items/tools/screwdriver.dm [hunk 1]
```diff
@@ -25,54 +25,42 @@
 	drop_sound = 'sound/items/handling/screwdriver_drop.ogg'
 	pickup_sound =  'sound/items/handling/screwdriver_pickup.ogg'
 	sharpness = SHARP_POINTY
-	var/random_color = TRUE //if the screwdriver uses random coloring
+	/// If the item should be assigned a random color
+	var/random_color = TRUE
+	/// List of possible random colors
 	var/static/list/screwdriver_colors = list(
-		"blue" = rgb(24, 97, 213),
```
Comment (ninjanomnom) [inline code/game/objects/items/tools/screwdriver.dm:30]: Is there some reason you changed from the rgb() format? These just output a color hex like you've manually done here.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/game/objects/items/tools/screwdriver.dm (hunk 1 above)
Comment (Arkatos1) [reply code/game/objects/items/tools/screwdriver.dm:30]: I just wanted to be consistent with other works and I also think RGB did not show color icon preview in VSC for me.

Comment (ninjanomnom) [reply code/game/objects/items/tools/screwdriver.dm:30]: ah fair enough

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**